pkg/azure/api/mock/mock_test.go

```diff
@@ -8,37 +8,28 @@ import (
 	"errors"
 	"testing"
 
-	check "github.com/cilium/checkmate"
+	"github.com/stretchr/testify/require"
 
 	"github.com/cilium/cilium/pkg/azure/types"
-	"github.com/cilium/cilium/pkg/checker"
 	"github.com/cilium/cilium/pkg/cidr"
 	ipamTypes "github.com/cilium/cilium/pkg/ipam/types"
 )
 
-func Test(t *testing.T) {
-	check.TestingT(t)
-}
-
-type MockSuite struct{}
-
-var _ = check.Suite(&MockSuite{})
-
-func (e *MockSuite) TestMock(c *check.C) {
+func TestMock(t *testing.T) {
 	subnet := &ipamTypes.Subnet{ID: "s-1", CIDR: cidr.MustParseCIDR("10.0.0.0/16"), AvailableAddresses: 65534}
 	api := NewAPI([]*ipamTypes.Subnet{subnet}, []*ipamTypes.VirtualNetwork{{ID: "v-1"}})
-	c.Assert(api, check.Not(check.IsNil))
+	require.NotNil(t, api)
 
 	instances, err := api.GetInstances(context.Background(), ipamTypes.SubnetMap{})
-	c.Assert(err, check.IsNil)
-	c.Assert(instances.NumInstances(), check.Equals, 0)
+	require.NoError(t, err)
+	require.Equal(t, 0, instances.NumInstances())
 
 	vnets, subnets, err := api.GetVpcsAndSubnets(context.Background())
-	c.Assert(err, check.IsNil)
-	c.Assert(len(vnets), check.Equals, 1)
-	c.Assert(vnets["v-1"], checker.DeepEquals, &ipamTypes.VirtualNetwork{ID: "v-1"})
-	c.Assert(len(subnets), check.Equals, 1)
-	c.Assert(subnets["s-1"], checker.DeepEquals, subnet)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(vnets))
+	require.Equal(t, &ipamTypes.VirtualNetwork{ID: "v-1"}, vnets["v-1"])
+	require.Equal(t, 1, len(subnets))
+	require.Equal(t, subnet, subnets["s-1"])
```

Comment:
nit: does this need to be `EqualValues`?

Reply:
subnet appears to be a single `ipamTypes.Subnet`, I think a deep-equals is sufficient?